lib/modules/platform/github/index.ts

```diff
@@ -550,108 +546,14 @@ export async function getRepoForceRebase(): Promise<boolean> {
   return config.repoForceRebase;
 }
 
-async function getClosedPrs(): Promise<PrList> {
-  if (!config.closedPrList) {
-    config.closedPrList = {};
-    try {
-      // prettier-ignore
-      const nodes = await githubApi.queryRepoField<GhGraphQlPr>(
-        closedPrsQuery,
-        'pullRequests',
-        {
-          variables: {
-            owner: config.repositoryOwner,
-            name: config.repositoryName,
-          },
-        }
-      );
-      const prNumbers: number[] = [];
-      // istanbul ignore if
-      if (!nodes?.length) {
-        logger.debug('getClosedPrs(): no graphql data');
-        return {};
-      }
-      for (const pr of nodes) {
-        // https://developer.github.com/v4/object/pullrequest/
-        pr.displayNumber = `Pull Request #${pr.number}`;
-        pr.state = pr.state.toLowerCase();
-        pr.sourceBranch = pr.headRefName;
-        delete pr.headRefName;
-        pr.comments = pr.comments.nodes.map((comment) => ({
-          id: comment.databaseId,
-          body: comment.body,
-        }));
-        pr.body = 'dummy body'; // just in case
-        config.closedPrList[pr.number] = pr;
-        prNumbers.push(pr.number);
-      }
-      prNumbers.sort();
-      logger.debug({ prNumbers }, 'Retrieved closed PR list with graphql');
-    } catch (err) /* istanbul ignore next */ {
-      logger.warn({ err }, 'getClosedPrs(): error');
-    }
-  }
-  return config.closedPrList;
+async function getClosedPrs(): Promise<Pr[]> {
+  const prList = await getPrList();
+  return prList.filter((pr) => pr.state === 'closed' || pr.state === 'merged');
```

Comment:
`state` is either `open` or `closed`. Merged PRs have `state` set to `closed` and `merged_at` and `merged_by` non-null.

Reply:
For some reason, here we split this `closed` into `closed` and `merged`:

https://github.com/renovatebot/renovate/blob/38de02e753c98f96ea375455c387e7b0f3069d9d/lib/modules/platform/github/index.ts#L637-L652

Reply:
Yeah, internally we differentiate between open, closed and merged. Now would not be the time to refactor

Reply:
Ah, my mistake. I thought this was the external API 🤦